### PR TITLE
altered DID Document check according to RFC006

### LINKED ISSUE #261: Change DID creation validation
A verificationMethod is no longer required. The only check that needs to be done is that the JWK in the transaction must match the fingerprint/DID of the new document.

### PR DESCRIPTION
**Pull request by woutslakhorst**
closes #261

### CODE REVIEW

File: vdr/doc/creator.go [hunk 1]
```diff
@@ -47,28 +43,22 @@ type Creator struct {
...
 
 	// generate idString
-	pkBytes := elliptic.Marshal(ecPKey.Curve, ecPKey.X, ecPKey.Y)
-	pkHash := sha256.Sum256(pkBytes)
-	idString := base58.Encode(pkHash[:], base58.BitcoinAlphabet)
-
-	// generate kid fragment
 	jwKey, err := jwk.New(pKey)
+	if err != nil {
+		return "", fmt.Errorf("could not generate kid: invalid key type")
```
Comment (reinkrul) [inline vdr/doc/creator.go:52]: why not just wrap `err`? It's now thrown away

File: vdr/doc/creator.go [hunk 2]
```diff
@@ -47,28 +43,22 @@ type Creator struct {
...
+	}
+
+	pkHash, err := jwKey.Thumbprint(crypto.SHA256)
 	if err != nil {
 		return "", err
 	}
+	idString := base58.Encode(pkHash[:], base58.BitcoinAlphabet)
```
Comment (reinkrul) [inline vdr/doc/creator.go:59]: make a const of `BitcoinAlphabet` since it's now used twice as JWK thumbprint encoding scheme (or make a util func)

**Review by reinkrul — CHANGES_REQUESTED** [review]
Ouch, there go all our development network DIDs :-p

File: vdr/doc/creator.go (hunk 2 above)
Comment (woutslakhorst) [reply vdr/doc/creator.go]: moved Thumbprint to crypto and added test that matches example from RFC

**Review by reinkrul — APPROVED** [review]
Seal of approval

```
          -~~~~-
         |O __ O|
      -__|_\__/_|__-
      ---(__/\__)---
        /||~--~||\
       / ||    || \
      |  `'    `'  |
    _/ /\__    __/\ \_
  ~~~~~    ~~~~    ~~~~~
```

**Pull request merged**